apps/jobs/views.py

```diff
@@ -332,11 +332,22 @@ def leaderboard(request, challenge_phase_split_id):
     # while populating leaderboard
     challenge_obj = challenge_phase_split.challenge_phase.challenge
     challenge_hosts_emails = challenge_obj.creator.get_all_challenge_host_email()
+    is_challenge_phase_public = challenge_phase_split.challenge_phase.is_public
+
+    if not is_challenge_phase_public:
+        challenge_hosts_emails = []
```

Comment:
If challenge phase is private we would not want the hosts to be excluded from leaderboard data, which is why we have to clear host emails for such cases. Otherwise the submissions will be filtered out based on email ids in next step